### PR TITLE
Transformers detector

### PR DESCRIPTION
**Pull request by alex-jw-brooks**
- Fixes bug in object detector data model schema
- Adds inference support for object detectors in transformers (tested with YOLOs & DETR); fine-tuning to follow in a future PR
- Adds tiny Yolo model for testing - generated using transformers utils from the library's test class, similar to how we do things in caikit NLP
- Updates dependencies to add YOLOs transformer support

### CODE REVIEW

File: caikit_computer_vision/modules/object_detection/transformers.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    ) -> "TransformersObjectDetector":
+        """Initialize a caikit-wrapped object detector."""
+        super().__init__()
+        self.model_name = model_name
+        self.image_processor = image_processor
+        self.detector_model = detector_model
+
+    @classmethod
+    def load(cls, model_path):
+        model_path = os.path.abspath(model_path)
```
Comment (gkumbhat) [inline caikit_computer_vision/modules/object_detection/transformers.py:68]: recently we have been started to move away from considering model_path strictly as `str` object, having this `abspath` might create problems in future. Can we instead just directly use `Moduleconfig.load` function?

File: pyproject.toml [hunk 2]
```diff
@@ -14,7 +14,10 @@ classifiers=[
     "License :: OSI Approved :: Apache Software License"
 ]
 dependencies = [
-    "caikit[runtime-grpc,runtime-http]>=0.16.0,<0.17.0",
+    "caikit[runtime-grpc,runtime-http,interfaces-vision]>=0.16.0,<0.17.0",
+    "transformers>=4.27.1,<5",
+    "torch>2.0,<3",
+    "timm>=0.9.5,<1"
```
Comment (gkumbhat) [inline pyproject.toml:20]: whats `timm` for?

**Review by gkumbhat — COMMENTED** [review]
Superb Alex. Left some small comments / questions

File: pyproject.toml (hunk 2 above)
Comment (alex-jw-brooks) [reply pyproject.toml:20]: It's a deep learning library that's used for some computer vision models! Some models, e.g., DEtection TRansformer require it to run

File: caikit_computer_vision/modules/object_detection/transformers.py (hunk 1 above)
Comment (alex-jw-brooks) [reply caikit_computer_vision/modules/object_detection/transformers.py]: Done 😄

**Review by gkumbhat — APPROVED** [review]
LGTM. Thanks Alex 🥳

**Pull request merged**